Se 13419 rtm infinite link loop hn (#216)
mulesoft/docs-runtime-manager: modules/ROOT/pages/servers-create.adoc

= Add Servers to Runtime Manager (Hybrid)
ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]

image:logo-cloud-disabled.png[link="/runtime-manager/deployment-strategies", title="CloudHub"]
image:logo-hybrid-active.png[link="/runtime-manager/deployment-strategies", title="Hybrid Deployment"]
image:logo-server-active.png[link="/runtime-manager/deployment-strategies", title="Anypoint Platform Private Cloud Edition"]

To add a Mule server to the Runtime Manager console so that you can manage it, you must first register it with the Runtime Manager agent. Use the `amc_setup` script to configure Runtime Manager agent to communicate with Runtime Manager. 

== Prerequisites

* Your enterprise license is current. 

> * You are running Mule 3.6.0 and later, and API gateway 2.1 and later.

Reviewer: ```suggestion
* You are running Mule 3.6.0 or later, and API gateway 2.1 or later.
```

* If you must download the agent, you have an Enterprise support account.

> * If the server is already registered with another Runtime Manager instance, remove it first.

Reviewer: Does the user remove the server from the instance or remove the registration of the server? Or are they really just the same thing and I am splitting hairs?
```suggestion
* If the server is already registered with another Runtime Manager instance, remove that registration first.
```

== Add a Server

To add a Mule server to Runtime Manager:

. From Anypoint Platform, select *Runtime Manager*.
. Click *Servers* in the left menu.
. Click the *Add Server* button.
+

> Runtime Manager generates the `amc_setup` command, including `_token_` and `_server-name_`, to use to register Mule in your environment:

Reviewer: ```suggestion
Runtime Manager generates the `amc_setup` command, including `_token_` and `_server-name_`, for you to use to register Mule in your environment:
```

+
.The arrow shows the `amc_setup` command generated by *Add Server*.
image::rtm-add-server.png[amc_setup command]
+
The registration token includes your organization's ID and the current environment.
. Click *Copy* to copy the command.
. In a terminal window, change to the `$MULE_HOME/bin` directory for the Mule instance that you're registering.
. Paste the command on the command line.
. Change the instance name `server-name` to the unique name to use to label Mule in the Runtime Manager console.
+

> The `server-name` value is unique across environments. The `server-name` can be the same for the same organization in different environments.

Reviewer: Would this be too confusing?
```suggestion
The `server-name` value is unique across environments, but it can be the same for one organization that is part of different environments.
```

Author: This makes it wrong.

. Include any other parameters on the `amc_setup` command line.
+
For information, see xref:installing-and-configuring-runtime-manager-agent.adoc#amc_setup-parameters[amc_setup Parameters].
. If your environment requires all outbound calls to go through a proxy, specify proxy settings in either the `$MULE_HOME/conf/mule-agent.yml` file or the `$MULE_HOME/conf/wrapper.conf` file.
+
See xref:rtm-agent-proxy-config.adoc[Connect the Agent Through a Proxy Server].
. If you are using the IBM JVM, see <<IBM JVM>>.

When the `amc_setup` command completes successfully, you see:

----
Mule Agent configured successfully
Connecting to Access Management to extract client_id and client_secret
Credentials extracted correctly, updating wrapper conf file
----

> After the script completes successfully, your server is listed on the *Servers* tab of Runtime Manager with a status of *Created*.

Reviewer: ditto status

If the server was running when you ran the `amc_setup` script, restart the server to reconnect with Runtime Manager.

== IBM JVM

When creating a server using the IBM Java Virtual Machine (JVM), you must use a different truststore file than the default truststore installed by Runtime Manager agent.

> . Download the custom truststore from this knowledge article:

Reviewer: ```suggestion
. Download the custom truststore from this article:
```
OR
```suggestion
. Download the custom truststore from this knowledgebase article:
```

+
https://support.mulesoft.com/s/article/Connectivity-issues-between-Mule-Runtime-and-Anypoint-Runtime-manager-using-an-IBM-JVM[Connectivity issues between Mule and Anypoint Runtime Manager using an IBM JVM]

> . Rename the `truststore` file in `$MULE_HOME/conf` to one of these file names, depending on the Mule version:

Reviewer: ```suggestion
. Depending on your Mule version, rename the `truststore` file in `$MULE_HOME/conf` to one of these file names:
```

* `anypoint-truststore.jks` - Mule 4.1.3 (Mule Agent 2.1.4) and later

> * `truststore.jks` — Mule 4.1.2 (Mule Agent 2.1.3)

Reviewer: GLOBALLY replace space-hyphen-space as a divider.
Consider a colon, parens, or a second line (the preferred format when the second part is longer than a phrase).
